roles/oneview_appliance_network_interfaces/tasks/main.yml

```diff
@@ -0,0 +1,41 @@
+---
+- name: Creates Network Interface if doesn't exist
+  oneview_appliance_network_interfaces:
+    config: "{{ config }}"
+    state: present
+    data:
+      macAddress: "{{mac_address}}"
+      ipv4Type: "STATIC"
+      ipv6Type: "UNCONFIGURE"
+      hostname: "{{hostname}}"
+      app1Ipv4Addr: "{{app1_ipv4_address}}"
+      app2Ipv4Addr: "{{app2_ipv4_address}}"
+      virtIpv4Addr: "{{ ipv4_address }}"
+      ipv4Subnet: "{{ ipv4_subnet }}"
+      ipv4Gateway: "{{ gateway }}"
+      ipv4NameServers: 
+         - "{{ dnsServer1 }}"
+         - "{{ dnsServer2 }}"
+  delegate_to: localhost
+- debug: var=appliance_network_interfaces
+
+- name: Do nothing when network interface exists
+  oneview_appliance_network_interfaces:
+    config: "{{ config }}"
+    state: present
+    data:
+      macAddress: "{{mac_address}}"
```

Comment:
try with entire data to check the corner idempotency issues

Reply:
I have already tested with all the params above for idempotency issues. I gave mac_address just to keep task information minimal.

Reply:
ok. But please add entire data in example.